lib/grab-module-data.js

```diff
@@ -3,56 +3,58 @@
 const createOptions = require('./create-options');
 const spawn = require('./spawn');
 
-function grabModuleData(context, next) {
-  // Launch npm info as a child process, using the tmp directory
-  // As the working directory.
-  const proc = spawn(
-    'npm',
-    ['view', '--json', context.module.raw],
-    createOptions(context.path, context)
-  );
-  let data = '';
-  proc.stdout.on('data', (chunk) => {
-    data += chunk;
-  });
-  proc.on('error', (err) => {
-    next(err);
-  });
-  proc.on('close', (code) => {
-    if (code > 0) {
+function grabModuleData(context) {
+  return new Promise((resolve, reject) => {
+    // Launch npm info as a child process, using the tmp directory
+    // As the working directory.
+    const proc = spawn(
+      'npm',
+      ['view', '--json', context.module.raw],
+      createOptions(context.path, context)
+    );
+    let data = '';
+    proc.stdout.on('data', (chunk) => {
+      data += chunk;
+    });
+    proc.on('error', (err) => {
+      reject(err);
+    });
+    proc.on('close', (code) => {
+      if (code > 0) {
+        context.emit(
+          'data',
+          'silly',
+          `${context.module.name} npm-view`,
+          'No npm package information available'
+        );
+        if (
+          context.module.type === 'git' &&
+          context.module.hosted.type === 'github'
+        ) {
+          context.meta = {
+            repository: {
+              type: 'git',
+              url: context.module.hosted.git()
+            }
+          };
+        }
+        resolve();
```

Comment:
Should we keep on returning the context? This function will now only work due to side effects and using the return value instead of waiting for the function to change the input object seems nicer to me.